code/core/src/main/java/com/adobe/marketing/mobile/internal/util/MapExtensions.kt

```diff
@@ -28,8 +28,10 @@ internal fun Map<String, Any?>.fnv1a32(masks: Array<String>? = null): Long {
     if (innerMasks?.isEmpty() == true) innerMasks = null
     innerMasks?.let {
         it.sortedArray().forEach { mask ->
-            if (mask.isNotEmpty() && flattenedMap.containsKey(mask)) {
-                kvPairs.append(mask).append(":").append(flattenedMap[mask].toString())
+            val mapValue = flattenedMap[mask].toString()
+            // we want to ignore null or empty values within the map
+            if (mask.isNotEmpty() && mapValue != "null" && mapValue != "") {
```

Comment:
Can you check for null before converting it to String and add some unit test case to cover this case?